src/main/kotlin/no/nb/bikube/core/util/DateUtils.kt

```diff
@@ -3,15 +3,20 @@ package no.nb.bikube.core.util
 import java.time.LocalDate
 
 val yearRegex = Regex("^\\d{4}$")
-val dateRegex = Regex("^\\d{4}-\\d{2}-\\d{2}$")
+val dateWithSlashRegex = Regex("^\\d{4}-\\d{2}-\\d{2}$")
+val dateWithDotsRegex = Regex("^\\d{4}\\.\\d{2}\\.\\d{2}$")
 
 class DateUtils {
     companion object {
         fun parseYearOrDate(date: String?): LocalDate? {
-            return if (date!!.matches(yearRegex)) {
+            return if (date.isNullOrBlank()) {
+                null
+            } else if (date.matches(yearRegex)) {
                 LocalDate.parse("$date-01-01")
-            } else if (date!!.matches(dateRegex)) {
+            } else if (date.matches(dateWithSlashRegex)) {
                 LocalDate.parse(date)
+            } else if (date.matches(dateWithDotsRegex)){
+                LocalDate.parse(date.replace(".", "-"))
             } else {
                 logger().warn("Could not parse date: $date")
                 null
```

Comment:
Could we possibly get dates delimited by "/", e.g. 2023/01/01? If so we should have a regex+condition for it.

Reply:
Not at the moment. GUI-syntax is always 2020-01-01 or 2020, and the dots are only to be able to parse from name (until the date field is actually added in Collections). I think we should await actual cases before adding support for more.